src/java/org/apache/cassandra/config/GuardrailsOptions.java

```diff
@@ -709,6 +735,16 @@ private static void validateMinRFThreshold(int warn, int fail, String name)
         validateMinRFVersusDefaultRF(fail, name);
     }
 
+    private static void validateMaxRFThreshold(int warn, int fail, String name)
+    {
+        validateMaxIntThreshold(warn, fail, name);
+        if (fail != -1 && fail < DatabaseDescriptor.getDefaultKeyspaceRF())
+        {
+            throw new IllegalArgumentException(String.format("%s_fail_threshold to be set (%d) cannot be lesser than default_keyspace_rf (%d)",
+                                                             name, fail, DatabaseDescriptor.getDefaultKeyspaceRF()));
+        }
+    }
+
```

Comment:
I would also inline `validateMinRFVersusDefaultRF` into `validateMinRFThreshold`, to keep consistency between these mostly symmetrical guardrails. Also, we can use `format` instead of `String.format` since we already have a static import for it. And we can also inline the `name` arguments, since they always have the same value, and try not to exceed the line width:
```java
private static void validateMinRFThreshold(int warn, int fail)
{
    validateMinIntThreshold(warn, fail, "maximum_replication_factor");

    if (fail > DatabaseDescriptor.getDefaultKeyspaceRF())
        throw new IllegalArgumentException(format("minimum_replication_factor_fail_threshold to be set (%d) " +
                                                  "cannot be greater than default_keyspace_rf (%d)",
                                                  fail, DatabaseDescriptor.getDefaultKeyspaceRF()));
}

private static void validateMaxRFThreshold(int warn, int fail)
{
    validateMaxIntThreshold(warn, fail, "maximum_replication_factor");

    if (fail != -1 && fail < DatabaseDescriptor.getDefaultKeyspaceRF())
        throw new IllegalArgumentException(format("maximum_replication_factor_fail_threshold to be set (%d) " +
                                                  "cannot be lesser than default_keyspace_rf (%d)",
                                                  fail, DatabaseDescriptor.getDefaultKeyspaceRF()));
}
```

Reply:
will fix the MinimumRF methods in a different commit